ring_theory/euclidean_domain.lean

```diff
@@ -0,0 +1,51 @@
+/-
+Copyright (c) 2018 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Chris Hughes
+-/
+import algebra.euclidean_domain ring_theory.ideals ring_theory.associated
+noncomputable theory
+local attribute [instance] classical.dec
+open euclidean_domain set ideal
+
+theorem span_gcd {α} [euclidean_domain α] (x y : α) :
```

Comment:
I had a bit of a discussion with @ChrisHughes24 about this result and the ones after it. Technically these lemmas are true in greater generality... (I guess in gcd domains). However the type class hierarchy in Lean is not as easy as the one on wiki (see https://en.wikipedia.org/wiki/GCD_domain). Therefore it seems that (for now) this might be a natural place to prove these facts.

Reply:
so should we base Euclidean domain on GCD domain?